### PR TITLE
Add article on Azure DevOps service connection security

### PR DESCRIPTION
**Pull request by BlueBasher**
# Pull Request Template

## What are you trying to address
This PR adds an article that describes a potential security issue in Azure DevOps and how to fix that issue.

## Description of new changes

A new article has been added that describes a security issue that can be present in an Azure DevOps setup regarding Service Connections. The article describes what the actual issue is and how Service Connections should be configured to fix the issue.

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [ ] Label the pull request with the appropriate area(s)
- [ ] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [ ] Assign the project - Engineering Playbook Backlog
- [ ] Assign the pull request to yourself

### CODE REVIEW

**Comment by TessFerrandez** [comment]
Thank you for the PR - generally I think the content is good, but I think perhaps it fits better in Continuous Delivery or Continuous Integration under dev-sec-ops -- there are some similar patterns in there

**Comment by BlueBasher** [comment]
Before creating the PR, I was in doubt whether it should be Security or Continuous Integration. But CI seems like a good fit indeed.
I've moved the content and updated the PR.

File: docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Azure DevOps Service Connection Security
+
+In Azure DevOps, Service Connections are used in Azure DevOps Pipelines to connect to external services, like Azure, GitHub, Docker, Kubernetes, and many other services. Service Connections can be used to authenticate to these external services and to invoke diverse types of commands, like create and update resources in Azure, upload container images to Docker, or deploy applications to Kubernetes.
+
+To be able to invoke these commands, Service Connections need to have the right permissions to do so, for most types of Service Connections the permissions can be scoped to a subset of resources to limit the access they have. To improve the the principle of least privilege, it's often very common to have separate Service Connections for different environments like Dev/Test/QA/Prod.
```
Comment (shiranr) [inline docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md:5]:
```suggestion
To be able to invoke these commands, Service Connections need to have the right permissions to do so, for most types of Service Connections the permissions can be scoped to a subset of resources to limit the access they have. To improve the principle of least privilege, it's often very common to have separate Service Connections for different environments like Dev/Test/QA/Prod.
```

File: docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Because Service Connections have all the necessary permissions in these external services, it is crucial to secure Service Connections so they cannot be misused by accident or by malicious users.
```
[the remark this answers is not in the document]
Comment (BlueBasher) [reply docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md]: Thanks, good point. I've moved the section and added an example Pipeline of how the Service Connection could be misused.

**Review by shiranr — APPROVED** [review]

**Review by shiranr — APPROVED** [review]
LGTM, thank you for contributing.

File: docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Azure DevOps Service Connection Security
+
+In Azure DevOps, Service Connections are used in Azure DevOps Pipelines to connect to external services, like Azure, GitHub, Docker, Kubernetes, and many other services. Service Connections can be used to authenticate to these external services and to invoke diverse types of commands, like create and update resources in Azure, upload container images to Docker, or deploy applications to Kubernetes.
```
Comment (TessFerrandez) [inline docs/continuous-integration/dev-sec-ops/azure-devops/service-connection-security.md:3]:
```suggestion
Service Connections are used in Azure DevOps Pipelines to connect to external services, like Azure, GitHub, Docker, Kubernetes, and many other services. Service Connections can be used to authenticate to these external services and to invoke diverse types of commands, like create and update resources in Azure, upload container images to Docker, or deploy applications to Kubernetes.
```

**Pull request merged**